system-int-test/test/cmr/system_int_test/ingest/service_ingest_test.clj

```diff
@@ -206,3 +207,51 @@
               {:keys [status concept-id revision-id]} response]
           (is (= 200 status))
           (is (= 3 revision-id)))))))
+
+(deftest related-url-type-and-subtype-check
+  (testing
+    "Check valid and invalid related type and subtypes. The first URL is good
+    while the second is bad. The error response should represent this showing
+    the second URL type failed to match a valid keyword."
+    (let [related-urls [{"URL" "https://example.gov/good1"
+                         "URLContentType" "PublicationURL"
+                         "Type" "USE SERVICE API"
+                         "Subtype" "OpenSearch"}  ;this is the valid Related URL
+                        {"URL" "https://example.gov/bad1"
+                         "URLContentType" "PublicationURL"
+                         "Type" "USE SERVICE APIs"
+                         "Subtype" "Closed Search"}
+                        {"URL" "https://example.gov/bad2"
+                         "URLContentType" "PublicationURL"
+                         "Type" "USE SERVICE APIs"
+                         "Subtype" "Very Closed Search"}]
+          concept-src (service-util/make-service-concept)
+          concept-result (as-> concept-src intermediate
+                               (json/parse-string (:metadata intermediate))
+                               (assoc intermediate "RelatedURLs" related-urls)
+                               (json/generate-string intermediate)
+                               (assoc concept-src :metadata intermediate)
+                               (ingest/ingest-concept intermediate))
+          {:keys [status errors]} concept-result]
+      (is (= 400 status))
+      (is (= 2 (count errors)))
+
+      (are3 [expected-type expected-subtype index error-item]
```

Comment:
Very clever test, using the previous results as expected cases for the `are3`. A bit complex, but covers all the cases I can think of nicely.

Reply:
in my first review, @sxu123 suggested that I add a second bad URL and confirm that I was getting multiple errors back. Since other tests confirm ingest, I did not think I needed to do that again. System liked the first keyword, and not the second two so yea, thought I was doing just enough to get by